src/renderers/shared/server/ReactPartialRenderer.js

```diff
@@ -487,6 +482,20 @@ class ReactDOMServerRenderer {
       if (child === null || child === false) {
         return '';
       } else {
+        var children = toArray(child);
```

Comment:
Let's not call `toArray()` unless we *know* this is not a single element. The case where single element is returned should be a fast one. For example `return <Foo />` shouldn't wrap `<Foo />` into array before traversing down.